src/main/kotlin/lotto/domain/LottoNumber.kt

```diff
@@ -1,13 +1,25 @@
 package lotto.domain
 
-data class LottoNumber(val value: Int) {
+import lotto.view.ExceptionMessage
+
+class LottoNumber private constructor(val value: Int) {
     init {
-        check(value in MINIMUM_NUMBER..MAXIMUM_NUMBER)
+        require(value in MINIMUM_NUMBER..MAXIMUM_NUMBER) { ExceptionMessage.REQUIRE_NUMBER_RANGE.message }
     }
 
     companion object {
         private const val MINIMUM_NUMBER = 1
         private const val MAXIMUM_NUMBER = 45
         val NUMBERS = MINIMUM_NUMBER..MAXIMUM_NUMBER
+        private val lottoNumberCache = mutableMapOf<Int, LottoNumber>()
+
+
+        fun from(value: Int): LottoNumber {
+            if (lottoNumberCache.containsKey(value)) {
+                return lottoNumberCache[value]!!
+            }
+            lottoNumberCache[value] = LottoNumber(value)
+            return lottoNumberCache[value]!!
+        }
```

Comment:
한번에 1~45의 로또 번호를 생성해둔 다음 제공하는 방식은 어떨까요?
```suggestion
        private val cache: Map<Int, LottoNumber> = NUMBERS.associateWith { LottoNumber(it) }
        fun from(value: Int): LottoNumber {
            return cache[value] ?: throw IllegalArgumentException()
        }
```